pkg/querier/tenantfederation/merge_metadata.go

```diff
@@ -48,13 +49,13 @@ func (m *mergeMetadataSupplier) MetricsMetadata(ctx context.Context) ([]scrape.M
 
 	if len(tenantIDs) == 1 {
 		level.Debug(spanlog).Log("msg", "only a single tenant, bypassing federated metadata supplier")
-		return m.next.MetricsMetadata(ctx)
+		return m.next.MetricsMetadata(ctx, opt)
 	}
 
 	results := make([][]scrape.MetricMetadata, len(tenantIDs))
 	run := func(jobCtx context.Context, idx int) error {
 		tenantID := tenantIDs[idx]
-		res, err := m.next.MetricsMetadata(user.InjectOrgID(jobCtx, tenantID))
+		res, err := m.next.MetricsMetadata(user.InjectOrgID(jobCtx, tenantID), opt)
 		if err != nil {
 			return fmt.Errorf("unable to run federated metadata request for %s: %w", tenantID, err)
 		}
```

Comment:
Do we need to do anything below to enforce `limit` and `limit_per_metric` as we merge results? Or will that be handled elsewhere? If the latter, might be good to add a comment explaining that.